### DISCUSSION TITLE
Possible to run SQL when libduckb is loaded?

### DISCUSSION THREAD

**Discussion by ramarnat** [duckdb/pg_duckdb#197] [Q&A] [answered]
Is it possible to run something like 
`CREATE SECRET s1 (TYPE AZURE, CONNECTION_STRING '$ENV_CONN_STRING')`
when duckdb is loaded, like a pre-hook?

**Comment by wuputah** [comment, marked as answer]
There's a diff here you could apply that would basically accomplish this, but we don't plan on shipping this since it would not fit within the "Postgres way" of managing things.
https://github.com/duckdb/pg_duckdb/pull/103/files

Proper Azure support ( #52 ) probably wouldn't be too hard, but it has a few edge cases to deal with.

**Comment by ramarnat** [comment]
I was looking at adding the azure support, and had a thought. Would it make more sense to have the secrets table as a key-value type table grouped by id, which would give it more flexibility in how the `CREATE SECRET` ddl is created?

**Comment by wuputah** [comment]
I did have a thought that a JSONB column might be a good idea, at least to replace any "sparse" columns... though I don't know how annoying it is to access the JSON data from C/C++.

**Reply by ramarnat** [reply]
Key-Value pair is probably simpler to implement for that reason...
